src/System.Private.CoreLib/src/System/Threading/Overlapped.cs

```diff
@@ -155,45 +156,44 @@ sealed internal unsafe class OverlappedData
 
     public class Overlapped
     {
-        private OverlappedData _overlappedData;
+        private OverlappedData? _overlappedData;
 
         public Overlapped()
         {
             // The split between Overlapped and OverlappedData should not be needed. It is required by the implementation of 
             // async GC handles currently. It expects OverlappedData to be a sealed type.
-            _overlappedData = new OverlappedData();
-            _overlappedData._overlapped = this;
+            _overlappedData = new OverlappedData(this);
         }
 
         public Overlapped(int offsetLo, int offsetHi, IntPtr hEvent, IAsyncResult ar) : this()
         {
-            _overlappedData.OffsetLow = offsetLo;
+            _overlappedData!.OffsetLow = offsetLo;
```

Comment:
Maybe it would be better to `Debug.Assert(_overlappedData != null);` to make all lines to look the same?

Reply:
Ok, I'll change that.